[Bug]: Test for parallel write failing with HDMF dev

### What happened?

The test suite is currently failing when running against the development version of HDMF on the `dev` branch.

https://github.com/hdmf-dev/hdmf/pull/953 added checks for initialization of ids in `DynamicTable`, which causes and error for the parallel tests because `ids` don't appear to be specified.

### Steps to Reproduce

```shell
pytest
```


### Traceback

```shell
======================================================================================== short test summary info =========================================================================================
FAILED tests/unit/test_parallel_write.py::test_simple_tqdm - ValueError: Cannot determine row id's for table. Must provide ids with same length as the columns when all columns are specified via DataChunkIterator objects.
FAILED tests/unit/test_parallel_write.py::test_compound_tqdm - ValueError: Cannot determine row id's for table. Must provide ids with same length as the columns when all columns are specified via DataChunkIterator objects.
============================================================================= 2 failed, 451 passed, 4002 warnings in 16.15s ==============================================================================
```


### Operating System

Windows

### Python Executable

Conda

### Python Version

3.8

### Package Versions

...

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf-zarr/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/hdmf-dev/hdmf-zarr/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,7 +6,7 @@
 * Enhanced ZarrIO to resolve object references lazily on read similar to HDMF's `HDF5IO` backend. @mavaylon1 [#120](https://github.com/hdmf-dev/hdmf-zarr/pull/120)
 
 ### New Features
-* Added parallel write support for the ``ZarrIO`` for datasets wrapped with ``GenericDataChunkIterator``. @CodyCBakerPhD [#118](https://github.com/hdmf-dev/hdmf-zarr/pull/118)
+* Added parallel write support for the ``ZarrIO`` for datasets wrapped with ``GenericDataChunkIterator``. @CodyCBakerPhD [#118](https://github.com/hdmf-dev/hdmf-zarr/pull/118) @oruebel [#128](https://github.com/hdmf-dev/hdmf-zarr/pull/128)
 
 ### Dependencies
 * Updated HDMF and PyNWB version to the most recent release @mavaylon1 [#120](https://github.com/hdmf-dev/hdmf-zarr/pull/120)

--- a/tests/unit/test_parallel_write.py
+++ b/tests/unit/test_parallel_write.py
@@ -189,7 +189,12 @@ def test_simple_tqdm(tmpdir):
                     display_progress=True,
                 )
             )
-            dynamic_table = DynamicTable(name="TestTable", description="", columns=[column])
+            dynamic_table = DynamicTable(
+                name="TestTable",
+                description="",
+                columns=[column],
+                id=list(range(3))  # must provide id's when all columns are iterators
+            )
             io.write(container=dynamic_table, number_of_jobs=number_of_jobs)
 
     assert expected_desc in tqdm_out.getvalue()
@@ -222,7 +227,10 @@ def test_compound_tqdm(tmpdir):
                 )
             )
             dynamic_table = DynamicTable(
-                name="TestTable", description="", columns=[pickleable_column, not_pickleable_column]
+                name="TestTable",
+                description="",
+                columns=[pickleable_column, not_pickleable_column],
+                id=list(range(3))  # must provide id's when all columns are iterators
             )
             io.write(container=dynamic_table, number_of_jobs=number_of_jobs)
 

